### ISSUE TITLE
[UX] Decrease size of help text input textareas

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#2747]
## Describe your issue or idea
All the "help text" textfields are huge. Can we please limit these to textfields instead of text areas?

To see an example of this part of the admin interface, use the admin menu to navigate to: Structure → Content types → Page → Manage fields → Body → Edit. Here's what it looks like:

![Screen Shot 2020-09-29 at 10 14 41 am](https://user-images.githubusercontent.com/2423362/94525416-bbfac800-023c-11eb-92b2-9e6fbec3d4dc.png)

## Current Proposal (as of Sept 29, 2020)

Reduce the size of the text area.

**Comment by stpaultim** [comment]
If I'm correct, it's a pretty simple PR to test this. I think we'd just need to change line 1917 of core/modules/field_ui/field_ui.admin.inc

Marking this "Good First Issue" to create a test PR. Recognizing that there may need to be some discussion of this change before this is implemented. If I'm correct, the PR would be pretty easy and provide an opportunity to see what this looks like and what effect it might have.

**Comment by klonos** [comment]
Yes, this would be a good first issue 👍 ...I've updated the issue summary with a screenshot + instructions on where to find this element in the admin interface.

Instead of decreasing the size, or making the text area a text field, may I propose that we add a checkbox and hide the help text area if that checkbox is not ticked? The same thing is done in the Views UI, and it provides a much betted UX in my opinion. To see what I mean, use the admin menu to navigate to: Structure → Views → Add view:

![Screen Shot 2020-09-29 at 10 17 45 am](https://user-images.githubusercontent.com/2423362/94525697-1d229b80-023d-11eb-9fef-459f2c6c96f8.png)

...if you click that button, you get a description field:

![Screen Shot 2020-09-29 at 10 18 58 am](https://user-images.githubusercontent.com/2423362/94525805-4d6a3a00-023d-11eb-8984-72637e89d554.png)

**Comment by indigoxela** [comment]
Please:

1. Do not make this a textfield, because changing the form item type would have impact on existing sites.
2. Do not complicate a simple task - it wouldn't be a "good first issue" anymore. (Fixing tests...)
3. Do not confuse volunteers with contradictory ideas and screenshots.

And please keep in mind that a change in UI might need discussion.

**Comment by klonos** [comment]
> Do not make this a textfield,

Are you proposing to close this issue, or reduce the number of lines in the text field? If the later, then what should it be? (currently set to 5 lines)

**Comment by indigoxela** [comment]
> Are you proposing to close this issue, or reduce the number of lines in the text field

Actually the latter. Two or three rows would make the textarea pretty compact - which is what @jenlampton initially was after.
And that would qualify this issue as "good first issue".

BTW: the issue description screenshot isn't really up to date. Our current form item description is much shorter.

**Comment by olafgrabienski** [comment]
Hm, now I'm confused. In the issue description I see the suggestion "textfields instead of text areas". Is this the original suggestion, or was it changed?

**Comment by indigoxela** [comment]
Yeah, @jenlampton's suggestion three years ago was to use textfields, but that's overdone and has impact on existing sites, whereas reducing the row count won't do any harm and still make the form item more compact.

It seems, in three years no one agreed with or opposed to this issue, so we can't assume consensus on a bigger change than the row number.

**Comment by klonos** [comment]
> ...reducing the row count won't do any harm and still make the form item more compact.

Well, I was thinking that if we were to fix this, we might as well fix it properly. With so much help text accompanying the field itself, reducing the number of rows won't help on mobile. Whereas if we had a toggle, we could hide the field and also most of that text. In hope that it will help us decide better, here's a side-by-side screenshot on mobile, showing current, reduced to 2 rows, and finally with a checkbox toggle:

![image](https://user-images.githubusercontent.com/2423362/94551261-bfeb1200-025d-11eb-92d4-02a0e91078be.png)

I personally don't feel that reducing the field to 2 or 3 rows is improving things. The user still has to scroll a great deal to see the rest of the settings or reach the form submit buttons, and what bothers me the most is that this field is not commonly used, so I think that it's unfair to be eating up all that vertical space.
